src/moonlink/src/storage/mooncake_table/snapshot.rs

```diff
@@ -1057,7 +1057,10 @@ impl SnapshotTableState {
         // Assert and update flush LSN.
         if let Some(new_flush_lsn) = task.new_flush_lsn {
             // Assert flush LSN doesn't regress, if not force snapshot.
-            if self.current_snapshot.data_file_flush_lsn.is_some() && !opt.force_create {
+            if self.current_snapshot.data_file_flush_lsn.is_some()
+                && !opt.force_create
+                && new_flush_lsn != 0
```

Comment:
If you take a look at the error message here: https://github.com/Mooncake-Labs/moonlink/issues/688
the failed assertion is not of 0 LSN
```sh
thread 'tokio-runtime-worker' panicked at moonlink/src/moonlink/src/storage/mooncake_table/snapshot.rs:1061:17:
assertion failed: `(left < right)`
  left: `2459562880`,
 right: `2459562880`
```